### PR TITLE
Added alerting rule for absence of MDS server

### PR DESCRIPTION
**Pull request by shtripat**
ceph-mixins-bug-id: ceph/ceph-mixins/issues/62
Signed-off-by: Shubhendu <shtripat@redhat.com>

### CODE REVIEW

**Comment by shtripat** [comment]
Needs verification.
DONT MERGE

**Comment by shtripat** [comment]
Verified the changes and works as expected.

File: alerts/absent_alerts.libsonnet [hunk 1]
```diff
@@ -38,6 +38,27 @@
...
+            alert: 'CephMdsIsAbsent',
+            expr: |||
+              absent(ceph_mds_metadata{%(cephExporterSelector)s, ceph_daemon=~"mds(.*)"} == 1)
+            ||| % $._config,
+            'for': $._config.mdsIsAbsentAlertTime,
+            labels: {
+              severity: 'warning',
+            },
+            annotations: {
+              message: 'Storage metadata service not available anymore.',
```
Comment (umangachapagain) [inline alerts/absent_alerts.libsonnet]: Message and Description are the same. Would be good to have more info in description or discard it.

Comment (shtripat) [reply alerts/absent_alerts.libsonnet]: Ack. Also as per comments from Paul, we will rather depend on count(ceph_mds_metadata) to alert if required replicas are available for MDS server.

Comment (shtripat) [reply alerts/absent_alerts.libsonnet]: Ack. Also as per comments from Paul, we will rather depend on count(ceph_mds_metadata) to alert if required replicas are available for MDS server.

File: extras/manifests/prometheus-ceph-rules.yaml [hunk 2]
```diff
@@ -56,13 +69,13 @@ spec:
     - alert: CephMonHighNumberOfLeaderChanges
       annotations:
         description: 'Ceph Monitor "{{ $labels.job }}": instance {{ $labels.instance
-          }} has seen {{ $value }} leader changes recently.'
+          }} has seen {{ $value }} leader changes per minute recently.'
```
Comment (umangachapagain) [inline extras/manifests/prometheus-ceph-rules.yaml]: Not a part of this PR.

File: extras/manifests/prometheus-ceph-rules.yaml [hunk 3]
```diff
@@ -56,13 +69,13 @@ spec:
...
         message: Storage Cluster has seen many leader changes recently.
         severity_level: warning
         storage_type: ceph
       expr: |
-        rate(ceph_mon_num_elections{job="rook-ceph-mgr"}[15m]) > 3
-      for: 30m
+        rate(ceph_mon_num_elections{job="rook-ceph-mgr"}[5m]) * 60 > 0.95
```
Comment (umangachapagain) [inline extras/manifests/prometheus-ceph-rules.yaml]: Same as above

File: alerts/absent_alerts.libsonnet [hunk 4]
```diff
@@ -38,6 +38,27 @@
...
+            expr: |||
+              sum(ceph_mds_metadata{%(cephExporterSelector)s} == 1) < 2
+            ||| % $._config,
+            'for': $._config.mdsMissingReplicasAlertTime,
+            labels: {
+              severity: 'warning',
+            },
+            annotations: {
+              message: 'Insufficient replicas for storage metadata service.',
+              description: 'Minimum required replicas for storage metadata service not available.',
```
Comment (umangachapagain) [inline alerts/absent_alerts.libsonnet]: Message and Description still give the exact same information.

**Review by umangachapagain — CHANGES_REQUESTED** [review]

File: extras/manifests/prometheus-ceph-rules.yaml (hunk 2 above)
Comment (shtripat) [reply extras/manifests/prometheus-ceph-rules.yaml]: May be it was not generated and pushed as part of earlier PR so comes here.

File: extras/manifests/prometheus-ceph-rules.yaml (hunk 3 above)
Comment (shtripat) [reply extras/manifests/prometheus-ceph-rules.yaml]: May be it was not generated and pushed as part of earlier PR so comes here.

File: alerts/absent_alerts.libsonnet (hunk 4 above)
Comment (shtripat) [reply alerts/absent_alerts.libsonnet]: Let me make it more meaningful.

Comment (shtripat) [reply alerts/absent_alerts.libsonnet]: Also meaning has to be same ;)

File: extras/manifests/prometheus-ceph-rules.yaml (hunk 2 above)
Comment (shtripat) [reply extras/manifests/prometheus-ceph-rules.yaml]: Should I send another PR to correct the same?

**Review by umangachapagain — APPROVED** [review]

**Pull request merged**